internal/installercache/installercache.go

```diff
@@ -1,61 +1,80 @@
 package installercache
 
 import (
+	"os"
 	"sync"
 
+	lru "github.com/hashicorp/golang-lru/v2"
 	"github.com/openshift/assisted-service/internal/oc"
 	"github.com/openshift/assisted-service/models"
 	"github.com/openshift/assisted-service/pkg/executer"
 	"github.com/openshift/assisted-service/pkg/mirrorregistries"
 	"github.com/sirupsen/logrus"
 )
 
-type installers struct {
-	sync.Mutex
-	releases map[string]*release
+//TODO: TwoQueueCache which support lfu policy does
+//      not support onEvict hook that we need in order
+//      to clear the files. Replace the cache when they
+//      do.
+type ReleaseCache struct {
+	releases *lru.Cache[string, *release]
+	log      logrus.FieldLogger
 }
 
 type release struct {
 	sync.Mutex
-	path string
+	Path string
 }
 
-var cache installers = installers{
-	releases: make(map[string]*release),
+func New(capacity int, log logrus.FieldLogger) *ReleaseCache {
+	rc := ReleaseCache{
+		log: log,
+	}
+	c, err := lru.NewWithEvict(capacity, rc.onEvicted)
+	if err != nil {
+		log.WithError(err).Errorln("Failed to initialize the release binaries cache")
+	}
+	rc.releases = c
+	return &rc
 }
 
-// Get returns a release resource for the given release ID
-func (i *installers) Get(releaseID string) *release {
-	i.Lock()
-	defer i.Unlock()
+func (i ReleaseCache) onEvicted(key string, r *release) {
+	if r == nil {
+		i.log.Errorf("Evicting entry %s with unexpected nil value", key)
+		return
+	}
+
+	r.Lock()
+	defer r.Unlock()
 
-	r, present := i.releases[releaseID]
-	if !present {
-		r = &release{}
-		i.releases[releaseID] = r
+	i.log.Infof("Evicting entry %s %v", key, r.Path)
+	err := os.Remove(r.Path)
```

Comment:
I wonder if just a one-time action of removing the release is enough here
I'm a bit concerned about leaking directories if we couldn't delete them on the ``onEvicted``, and the only "solution" I can think of is something that runs periodically and makes sure to GC those dirs

Anyway it's mainly a concern. We can see how much it's probable after letting it run for a while

Reply:
I agree we are leaking directories here but the main concern is the files and the space they take. In reality there are not that many versions running in parallel: 90% of the customers holds about 5 versions.

Reply:
yeah, I mean directories and files interchangeably. The concern here is leaking disk space because we only try to delete on a one-time basis